### ISSUE TITLE
Incompatible pointers in generated C code regarding asserts

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#8194]
Simulate Modelica.Blocks.Examples.PID_Controller with OMEdit. Two red lines show up during C compilation:
```
PID_Controller.c:1018:14: warning: incompatible function pointer types assigning to 'void (*)(threadData_t *, FILE_INFO, const char *, ...) __attribute__((noreturn))' (aka 'void (*)(struct threadData_s *, struct _FILE_INFO, const char *, ...) __attribute__((noreturn))') from 'void (threadData_t *, FILE_INFO, const char *, ...)' (aka 'void (struct threadData_s *, struct _FILE_INFO, const char *, ...)') [-Wincompatible-function-pointer-types]
  omc_assert = omc_assert_simulation;
             ^ ~~~~~~~~~~~~~~~~~~~~~
PID_Controller.c:1024:13: warning: incompatible function pointer types assigning to 'void (*)(threadData_t *) __attribute__((noreturn))' (aka 'void (*)(struct threadData_s *) __attribute__((noreturn))') from 'void (threadData_t *)' (aka 'void (struct threadData_s *)') [-Wincompatible-function-pointer-types]
  omc_throw = omc_throw_simulation;
            ^ ~~~~~~~~~~~~~~~~~~~~
```
I'm not sure if this thing is important, but it's not nice to see red warnings in the compile window. @AnHeuermann, is this related to your recent work in #6195? If not, any clue what could be causing it?

**Comment by perost** [comment]
I think this was fixed in #8182.

**Comment by AnHeuermann** [comment]
>  @AnHeuermann, is this related to your recent work in #6195? If not, any clue what could be causing it?

No, that's older.

**Comment by AnHeuermann** [comment]
Already fixed in #8182

**Comment by casella** [comment]
I confirm the red stuff is gone with the latest nightly build, thanks! 👍
